### PR TITLE
feat(api): add lfs metadata

### PR DESCRIPTION
**Pull request by McPatate**
Full lfs object looks like this :
```json
"lfs": {
    "sha256": "5b1da7ebe3bdcc58c91eb9916f7d881791ac2659307f0c383c325361382f2f42",
    "size": 2239866697,
    "pointerSize": 135
}
```

Needed for the repo scanner

### CODE REVIEW

File: src/api/sync.rs [hunk 1]
```diff
@@ -501,12 +507,69 @@ impl ApiRepo{
...
+    /// # tokio_test::block_on(async {
+    /// let api = Api::new().unwrap();
+    /// api.model("gpt2".to_owned())
+    ///     .detailed_info()
+    ///     .query("blobs".to_owned(), "true".to_owned())
+    ///     .send();
+    /// # })
+    /// ```
+    pub fn detailed_info(&self) -> DetailedInfoBuilder {
+        DetailedInfoBuilder::new(self.api.clone(), &self.api.endpoint, &self.repo)
```
Comment (Narsil) [inline src/api/sync.rs:525]:
```suggestion
        DetailedInfoBuilder::new(self.api.clone(), &self.repo)
```
endpoint is already in api, no need to duplicate.

Comment (Narsil) [reply src/api/sync.rs:525]: Actually use `ApiRepo` instead so you need just Self.

File: src/api/sync.rs [hunk 2]
```diff
@@ -501,12 +507,69 @@ impl ApiRepo{
...
+        if let Some(qps) = self.query_params {
+            let qps: Vec<(&str, &str)> = qps
+                .iter()
+                .map(|qp| (qp.0.as_ref(), qp.1.as_ref()))
+                .collect();
+            request = request.query_pairs(qps);
+        }
+        let response = request.call().map_err(Box::new)?;
+
+        Ok(response.into_json()?)
```
Comment (Narsil) [inline src/api/sync.rs:572]:
```suggestion
impl DetailedInfoBuilder {
    fn new(api: &ApiRepo) -> Self {
        let url = format!("{}/api/{}", endpoint, repo.api_url());
         let mut request = self.api.client.get(&url);
        Self {
           request
        }
    }
}

impl DetailedInfoBuilder {
    /// Add a query parameter to the request
    pub fn query(mut self, key: String, value: String) -> Self {
        self.request = self.request.query(key, value)
        self
    }

    /// Send the request and get the resulting JSON
    pub fn send(self) -> Result<Value, ApiError> {
        let response = self.request.call().map_err(Box::new)?;
        Ok(response.into_json()?)
```

WDYT ?

File: src/api/sync.rs [hunk 3]
```diff
@@ -84,6 +85,10 @@ pub enum ApiError {
     /// We tried to download chunk too many times
     #[error("Too many retries: {0}")]
     TooManyRetries(Box<ApiError>),
+
+    /// serde_json error
+    #[error("Serde json error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
```
Comment (Narsil) [inline src/api/sync.rs:91]: How were we getting away without it before, is it the map(Box?) ?

File: src/api/tokio.rs [hunk 4]
```diff
@@ -591,14 +593,67 @@ impl ApiRepo{
...
+    pub async fn send(self) -> Result<Value, ApiError> {
+        let mut request = self.api.client.get(&self.url);
+        if let Some(qps) = self.query_params {
+            request = request.query(&qps);
+        }
+        let response = request.send().await?;
+        let response = response.error_for_status()?;
 
-        Ok(model_info)
+        Ok(response.json().await?)
```
Comment (Narsil) [inline src/api/tokio.rs:656]: Same API structure a proposed in Sync for simplicity.

**Review by Narsil — COMMENTED** [review]
Quite nice.

I propose a different (simpler imho) internals.

Otherwise looks good. Not sure how `cargo fmt` didn't pick up those before.

File: src/api/sync.rs (hunk 2 above)
Comment (Narsil) [reply src/api/sync.rs:572]: Maybe even Scratch the builder, and return the `Request` directly.. One less struct and full control over the querying to the user...

Comment (Narsil) [reply src/api/sync.rs:572]: Then the fn could be rename `info_request` maybe ?

File: src/api/sync.rs (hunk 3 above)
Comment (McPatate) [reply src/api/sync.rs:91]: `.into_json` returns an `io::Result<T>` while `serde_json::from_value` returns a `Result<T, serde_json::Error>`

Comment (McPatate) [reply src/api/sync.rs:91]: And `reqwest::Response::json` returns a `Result<T, reqwest::Error>`

File: src/api/sync.rs (hunk 2 above)
Comment (McPatate) [reply src/api/sync.rs:572]: Yep agree, `info_request` sounds good

Comment (McPatate) [reply src/api/sync.rs]: looks better now !

**Review by Narsil — APPROVED** [review]
LGTM

**Pull request merged**